diacritics_test.go

```diff
@@ -3,8 +3,55 @@ package main
 import (
 	"strconv"
 	"testing"
+	"unicode"
 )
 
+func Test_normMap(t *testing.T) {
+	t.Run("european", func(t *testing.T) {
+		europeanBase := []rune("ěřůøĉĝĥĵŝŭèùÿėįųāēīūļķņģőűëïąćęłńśźżõșțčďĺľňŕšťýžéíñóúüåäöçîşûğăâđêôơưáàãảạ")
+		europeanNorm := []rune("eruocghjsueuyeiuaeiulkngoueiacelnszzostcdllnrstyzeinouuaaocisugaadeoouaaaaa")
+
+		t.Run("lowercase", func(t *testing.T) {
+			for i := range europeanBase {
+				if normMap[europeanBase[i]] != europeanNorm[i] {
```

Comment:
`normMap` should be treated as an implementation detail, and can be tested by calling `removeDiacritics`. It seems like these tests duplicate the existing ones in `TestRemoveDiacritics` - while I think it's fine if you want to change the way `TestRemoveDiacritics` is written, I don't think it's necessary to have two levels of testing like this.

I would also prefer to hardcode the uppercase strings instead of using `unicode.ToUpper`, to keep as much logic out of the unit tests as possible.

Reply:
Yes, you are right, my test is redundant, so I removed it. I wanted to confirm my changes didn't break anything while writing it, but yes, the other test does the same thing